test/testdata/output/issue_55.golden

```diff
@@ -0,0 +1,15 @@
+name: sample
+version: 0.1.0
+environment:
+  sdk: ^3.0.0
+dependencies:
+dev_dependencies:
+  retry:
+---
+name: sample
+version: 0.1.0
+environment:
+  sdk: ^3.0.0
+dependencies:
+dev_dependencies:
+  {}
```

Comment:
I first had expected

```
dependencies:
dev_dependencies:
```

But I can totally see how that changes semantics.

I guess we handle that for `pub remove` in: https://github.com/dart-lang/pub/blob/master/lib/src/command/remove.dart#L158

And for `pub add` in https://github.com/dart-lang/pub/blob/e1fbda73d1ac597474b82882ee0bf6ecea5df108/lib/src/command/add.dart#L727.

Reply:
Yes, I think we should handle those special cases. Ideally we should consider removing the `dev_dependencies` key completely, when it's value is an empty map.

But as you point out: this is outside the scope of `yaml_edit`